wrangler-transform/src/main/java/io/cdap/wrangler/Precondition.java

```
@@ -72,9 +73,10 @@ public boolean apply(Row row) throws PreconditionException {
       scriptContext.setBindings(bindings, ScriptContext.ENGINE_SCOPE);
       Object result = script.eval(scriptContext);
       if (!(result instanceof Boolean)) {
-        throw new PreconditionException(
-          String.format("Precondition '%s' does not result in true or false.", condition)
-        );
+        String errorMessage = String.format("Precondition '%s' does not result in true or false.",
+            condition);
+        throw WranglerErrorUtil.getProgramFailureExceptionDetailsFromChain(null, errorMessage,
+            ErrorType.USER);
```

why this change?

Why can't we let it throw `PreConditionException`?